projects/components/src/table/cells/table-cell-renderer-lookup.service.ts

```diff
@@ -1,83 +1,60 @@
 import { Injectable, Injector } from '@angular/core';
-import { LoggerService } from '@hypertrace/common';
 import { TableColumnConfig, TableRow } from '../table-api';
 import {
-  TABLE_CELL_RENDERER_CELL_DATA,
-  TABLE_CELL_RENDERER_COLUMN_CONFIG,
-  TABLE_CELL_RENDERER_COLUMN_INDEX,
-  TABLE_CELL_RENDERER_ROW_DATA
+  TABLE_CELL_DATA,
+  TABLE_COLUMN_CONFIG,
+  TABLE_COLUMN_INDEX,
+  TABLE_ROW_DATA
 } from './table-cell-injection-tokens';
 import { TableCellRendererConstructor } from './table-cell-renderer';
 
 @Injectable({
   providedIn: 'root'
 })
 export class TableCellRendererLookupService {
-  private defaultRenderer?: TableCellRendererConstructor;
-
   private readonly renderers: Map<string, TableCellRendererConstructor> = new Map();
 
-  public constructor(private readonly loggerService: LoggerService) {}
-
-  public register(renderer: TableCellRendererConstructor, isDefault: boolean = false): void {
+  public register(renderer: TableCellRendererConstructor): void {
     this.renderers.set(renderer.type, renderer);
-
-    if (isDefault || this.renderers.size === 1) {
-      this.defaultRenderer = renderer;
-    }
   }
 
-  public registerAll(renderers: TableCellRendererConstructor[], defaultRenderer?: TableCellRendererConstructor): void {
+  public registerAll(renderers: TableCellRendererConstructor[]): void {
     renderers.forEach(renderer => {
-      this.register(renderer, renderer === defaultRenderer);
+      this.register(renderer);
     });
   }
 
-  public lookup(type: string | undefined): TableCellRendererConstructor {
-    if (type === undefined) {
-      return this.defaultOrError();
-    }
-
+  public lookup(type: string): TableCellRendererConstructor {
     if (!this.renderers.has(type)) {
-      this.loggerService.warn(`Table cell renderer by the name '${type}' not found. Using default.`);
-
-      return this.defaultOrError();
+      throw Error(`Table cell parser of type '${type}' not registered.`);
```

Comment:
No default any more?

Reply:
No. Default is set in the model now if nobody overrides it. Simplifies things a bit.

Reply:
Does it mean using the table directly anywhere outside a dashboard is required to be explicit?

Reply:
Hmm. Good point. We probably don't want to require that.

Going to change this back to have a default.

Reply:
Fixed.